includes/sensei-functions.php

```diff
@@ -281,14 +281,42 @@ function sensei_user_login_url() {
  * duplicate of Sensei()->access_settings().
  *
  * @since 1.9.0
+ * @since 3.5.2 Added hook to filter the return.
  * @return bool
  */
 function sensei_is_login_required() {
+	global $post;
 
-	$login_required = isset( Sensei()->settings->settings['access_permission'] ) && ( true == Sensei()->settings->settings['access_permission'] );
+	$post_type = get_post_type( $post );
+	$course_id = null;
 
-	return $login_required;
+	if ( 'course' === $post_type ) {
```

Comment:
We need to also handle the quiz case here in a switch.

Reply:
Good catch! Added the quiz check here: https://github.com/Automattic/sensei/pull/3631/commits/2bb7fb558b3d3129be5183c1d3ea0e565127b11e

And converted to a `switch`.